src/frontend/src/binder/relation/mod.rs

```diff
@@ -377,16 +378,23 @@ impl Binder {
                     Ok(Relation::BackCteRef(Box::new(BoundBackCteRef { share_id })))
                 }
                 BindingCteState::Bound { query } => {
+                    let schema = match query.clone() {
+                        Left(normal) => normal.body.schema().clone(),
+                        Right(recursive) => recursive.recursive.body.schema().clone(),
```

Comment:
same, could we just use the `recursive` schema here?

Reply:
plus, for the case like below (e.g., `<value> union all <select stmt>`),
we might directly use rhs's schema as the final schema.
```
with recursive t(n) as (
    values (1)
    union all
    select n + 1 from t where n < 100
)
select * from t;
```